python/lsst/analysis/tools/tasks/catalogMatch.py

```diff
@@ -0,0 +1,408 @@
+import numpy as np
+from astropy.time import Time
+from astropy.coordinates import SkyCoord
+import astropy.units as u
+import pandas as pd
+
+import lsst.pipe.base as pipeBase
+import lsst.pex.config as pexConfig
+import lsst.geom
+from lsst.meas.algorithms import ReferenceObjectLoader, LoadReferenceObjectsTask
+from lsst.pipe.tasks.configurableActions import ConfigurableActionStructField
+from lsst.skymap import BaseSkyMap
+
+from ..vectorActions.selectors import (CoaddPlotFlagSelector,
+                                       StellarSelector,
+                                       SnSelector,
+                                       GalacticSelector)
+
+__all__ = ["CatalogMatchConfig", "CatalogMatchTask", "AstropyMatchConfig", "AstropyMatchTask"]
+
+
+class AstropyMatchConfig(pexConfig.Config):
+
+    maxDistance = pexConfig.Field(
+        doc="Max distance between matches in arcsec",
+        dtype=float,
+        default=1.0,
+    )
+
+
+class AstropyMatchTask(pipeBase.Task):
+    """ A task for running the astropy matcher `match_to_catalog_sky` on
+    between target and reference catalogs."""
+
+    ConfigClass = AstropyMatchConfig
+
+    def run(self, refCatalog, targetCatalog):
+        """Run matcher
+
+        Parameters
+        ----------
+        refCatalog: `pandas.core.frame.DataFrame`
+            The reference catalog with coordinates in degrees
+        targetCatalog: `pandas.core.frame.DataFrame`
+            The target catalog with coordinates in degrees
+
+        Returns
+        -------
+        `pipeBase.Struct` containing:
+            refMatchIndices: `numpy.ndarray`
+                Array of indices of matched reference catalog objects
+            targetMatchIndices: `numpy.ndarray`
+                Array of indices of matched target catalog objects
+            separations: `astropy.coordinates.angles.Angle`
+                Array of angle separations between matched objects
+        """
+        refCat_ap = SkyCoord(ra=refCatalog['coord_ra'] * u.degree,
+                             dec=refCatalog['coord_dec'] * u.degree)
+
+        sourceCat_ap = SkyCoord(ra=targetCatalog['coord_ra'] * u.degree,
+                                dec=targetCatalog['coord_dec'] * u.degree)
```

Comment:
Should we have spaces around the * here?

Reply:
Do we know that the units are always gong to be degrees or should it be a config option?

Reply:
My mind is blown -- somehow I have been doing spacing around operators wrong. 

I added a config option for the units.

Reply:
When you switch the code to using `black` you will find that that rule about spaces around operators goes away anyhow...